server/src/termdb.gdc.status.js

```diff
@@ -0,0 +1,45 @@
+import path from 'path'
+import ky from 'ky'
+import serverconfig from '#src/serverconfig.js'
+
+/**
+ * Tests the GDC API status and returns true if the status is OK.
+ * If serverconfig.features.runRemainingWithoutAwait is false, it will throw an error if the status is not OK.
+ * @param ds
+ * @returns {Promise<boolean>}
+ */
+export async function testGDCApiStatus(ds) {
+	try {
+		const status = await getApiStatus(ds, 4)
+		if (status?.status !== 'OK') {
+			if (!serverconfig.features.runRemainingWithoutAwait) {
+				throw new Error('GDC API status is not OK')
+			} else {
+				return false
```

Comment:
I think this should automatically retry every 10 minutes or so, until the GDC API is back up, otherwise the GDC API will always be down in prp1 until the next restart/deployment. So, something like
```js
...
import { sleep } from '#src/utils'
...

const retries = {interval: 10 * 60 * 1000, max: 1000, total: 0}

export async function testGDCApiStatus(ds) {
	try {
		const status = await getApiStatus(ds, 4)
		if (status?.status !== 'OK') {
			if (!serverconfig.features.runRemainingWithoutAwait) {
				throw new Error('GDC API status is not OK')
			} else {
			        retries.total++
			         await sleep(retries.interval)
			         await testGDCApiStatus(ds)
			}
		}

```

Reply:
ah, just saw that the retry in in `getApiStatus()`. Why is this additional wrapper function needed?